### PR TITLE
Add new LXC: Gitea

### PR DESCRIPTION
**Pull request by Rogue-King**
## Description

Gitea is a painless, self-hosted, all-in-one software development service. It includes Git hosting, code review, team collaboration, package registry, and CI/CD. It is similar to GitHub, Bitbucket and GitLab.

Screenshot 2024-07-23 220826

## Application

Screenshot 2024-07-23 224602
These were the Install steps that were used: https://docs.gitea.com/installation/install-from-binary

## Type of change

- [X] New Script (Develop a new script or set of scripts that are fully functional and thoroughly tested)
- [X] I have performed a self-review of my code, adhering to established codebase patterns and conventions.
- [X] This change requires a documentation update

### CODE REVIEW

File: ct/gitea.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/tteck/Proxmox/main/misc/build.func)
+# Copyright (c) 2021-2024 tteck
+# Author: tteck (tteckster)
+# Co-author: Rogue-King
+# License: MIT
+# https://github.com/tteck/Proxmox/raw/main/LICENSE
+
+function header_info {
```
Comment (tteck) [inline ct/gitea.sh:9]: Capital G

File: ct/gitea.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  / __  / // __/ _ \/ __  /
+ / /_/ / // /_/  __/ /_/ /
+ \__, /_/ \__/\___/\__,_/
+/____/
+
+EOF
+}
+header_info
+echo -e "Loading..."
+APP="gitea"
```
Comment (tteck) [inline ct/gitea.sh]: Capital G

File: ct/gitea.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
+header_info
+if [[  ! -f /usr/local/bin/gitea ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+RELEASE=$(wget -q https://github.com/go-gitea/gitea/releases/latest -O - | grep "title>Release" | cut -d " " -f 4)
+VERSION=${RELEASE#v}
```
Comment (tteck) [inline ct/gitea.sh]: not needed

File: ct/gitea.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+msg_info "Updating gitea to ${RELEASE}"
+wget -q https://github.com/go-gitea/gitea/releases/download/$RELEASE/gitea-$VERSION-linux-amd64
```
Comment (tteck) [inline ct/gitea.sh:63]: `wget -q https://github.com/go-gitea/gitea/releases/download/$RELEASE/gitea-v${REALEASE}-linux-amd64`

File: install/gitea-install.sh [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
```
Comment (tteck) [inline install/gitea-install.sh:17]: missing `curl` (default base dependency)

File: install/gitea-install.sh [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+$STD apt-get install -y git
+$STD apt-get install -y sudo
+$STD apt-get install -y mc
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Gitea"
+RELEASE=$(wget -q https://github.com/go-gitea/gitea/releases/latest -O - | grep "title>Release" | cut -d " " -f 4)
+VERSION=${RELEASE#v}
```
Comment (tteck) [inline install/gitea-install.sh:25]: again, not needed

**Comment by tteck** [comment]
FYI, Gitea is available as a LXC in https://tteck.github.io/Proxmox/#turnkey-lxc-appliances

**Comment by tteck** [comment]
Since Gitea is already available via Turnkey, is there a need to maintain another script?

**Comment by SoTHISIsFitness** [comment]
> Since Gitea is already available via Turnkey, is there a need to maintain another script?

A lot of people really don't like Turnkey - common complaints include lack of updates / poor updates, which in and of itself is a nonstarter for many.

ETA: a quick google search will show quite a litany of issues / complaints regarding Turnkey. Your scripts are infinitely better and much simpler. Turnkey, as it turns out, isn't very 'turnkey' a lot of the time :/

**Comment by Rogue-King** [comment]
I have never used turnkey so didn't know it was available there but like SoTHISIsFitness mentioned your scripts are infinitely better and much simpler.

**Comment by Rogue-King** [comment]
@tteck Made the changes, and I tested the wget of the binary with the new modifications separately, it should work.

**Comment by tteck** [comment]
Did you make all changes?

Edit: I must have hit the preview button by mistake.

File: install/gitea-install.sh [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+msg_info "Installing Dependencies"
+$STD apt-get install -y git
+$STD apt-get install -y curl
+$STD apt-get install -y sudo
+$STD apt-get install -y mc
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Gitea"
+RELEASE=$(wget -q https://github.com/go-gitea/gitea/releases/latest -O - | grep "title>Release" | cut -d " " -f 4 | sed 's/^v//')
+$STD wget -q https://github.com/go-gitea/gitea/releases/download/v$RELEASE/gitea-$RELEASE-linux-amd64
```
Comment (tteck) [inline install/gitea-install.sh]: `$STD ` not needed

File: install/gitea-install.sh [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+mv gitea* /usr/local/bin/gitea
+chmod +x /usr/local/bin/gitea
+msg_info "Installed Gitea"
+
+msg_info "Creating Gitea user"
```
Comment (tteck) [inline install/gitea-install.sh]: add to "Installing Gitea" section. Make sure it's silent.

File: install/gitea-install.sh [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+adduser --system --group --disabled-password --home /etc/gitea gitea > /dev/null
+
+msg_info "Creating directory structure"
```
Comment (tteck) [inline install/gitea-install.sh]: add to "Installing Gitea" section.

**Comment by Rogue-King** [comment]
All changes are implemented, I don't know why the adduser command didn't silence before when it should have, but it works now and is tested:
swappy-20240725-115734

**Pull request merged**